### PR TITLE
Dynamic refresh rates for smart transactions

### PR DESCRIPTION
**Pull request by meppsilon**
Fixes: #

Explanation:  

Manual testing steps:  
  - 
  - 
  -

### CODE REVIEW

File: ui/ducks/swaps/swaps.js [hunk 1]
```diff
@@ -349,6 +349,21 @@ export const getLatestSmartTransactionUuid = (state) => {
   return state.appState.latestSmartTransactionUuid;
 };
 
+export const getSwapsRefreshStates = (state) => {
+  const {
+    swapsQuoteRefreshTime,
+    swapsQuotePrefetchingRefreshTime,
+    swapsStxGetTransactions,
```
Comment (dan437) [inline ui/ducks/swaps/swaps.js:356]: nit: from the `swapsStxGetTransactions` name is not immediately clear to me that it's a refresh rate / refresh time. Could we add a postfix `RefreshTime` to keep it consistent with the current refresh rates?

File: ui/pages/swaps/view-quote/view-quote.js [hunk 2]
```diff
@@ -191,7 +190,7 @@ export default function ViewQuote() {
       };
       intervalId = setInterval(() => {
         dispatch(fetchUnsignedTransactionsAndEstimates(unsignedTx));
-      }, FETCH_SMART_TRANSACTIONS_INTERVAL);
+      }, swapsRefreshRates.stxGetTransactions);
```
Comment (dan437) [inline ui/pages/swaps/view-quote/view-quote.js:193]: Nice!

File: ui/ducks/swaps/swaps.js (hunk 1 above)
Comment (meppsilon) [reply ui/ducks/swaps/swaps.js:356]: Done!

File: app/scripts/controllers/swaps.js [hunk 3]
```diff
@@ -168,6 +172,10 @@ export default class SwapsController {
           swapsRefreshRates?.quotes || FALLBACK_QUOTE_REFRESH_TIME,
         swapsQuotePrefetchingRefreshTime:
           swapsRefreshRates?.quotesPrefetching || FALLBACK_QUOTE_REFRESH_TIME,
+        swapsStxGetTransactionsRefreshTime:
```
Comment (dan437) [inline app/scripts/controllers/swaps.js:175]: I would have fallback for these two every 10 seconds, as it is now with hardcoded values.

**Review by dan437 — APPROVED** [review]

**Review by dan437 — APPROVED** [review]

**Pull request merged**